bin/weewx/drivers/vantage.py

```diff
@@ -1110,14 +1212,27 @@ def hardware_name(self):
     def archive_interval(self):
         return self.archive_interval_
     
+    def determine_hardware(self):
```

Comment:
If the objective is to repeat max_tries time, then breaking it out like this makes sense.

But, I'm asking why do it in the first place? Was it raising an error?